yt/fields/magnetic_field.py

```diff
@@ -170,35 +170,56 @@ def _mach_alfven(field,data):
 
 def setup_magnetic_field_aliases(registry, ds_ftype, ds_fields, ftype="gas"):
```

Comment:
@jzuhone I have merged the treatments for grid and particle dataset into a single function as you suggested. Let me know if you have further suggestions.